resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -432,6 +432,13 @@ private void query(final DnsServerAddressStream nameServerAddrStream,
         final Promise<AddressedEnvelope<? extends DnsResponse, InetSocketAddress>> queryPromise =
                 parent.ch.eventLoop().newPromise();
 
+        final long queryStartTimeNanos;
+        if (nameServerAddrStream instanceof DnsServerResponseFeedbackAddressStream) {
+            queryStartTimeNanos = System.nanoTime();
+        } else {
+            queryStartTimeNanos = -1;
```

Comment:
`System.nanoTime()` can return negative values. Use a separate boolean variable for tracking whether timing is enabled.

Reply:
Good feedback from a correctness point of view. I was interested and looked at the various OS implementations and wasn't able to find an example where this would happen but it is very much allowed per the (loosely) defined spec for `CLOCK_MONOTONIC`, fortunately `RDTSC` does define that the rollover is to zero.

Thanks!